src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            var data = e.target.result;
+            var path = file.name; // TODO: to a new directory?
```

Comment:
What kind of filenames do you see here? I presume `file.name` is the name of the file without any paths? So e.g. dragging a file `c:\path\to\file.txt` would show `file.txt` as `file.name` here? In that case, using that as a path directly would mean this creates a new file in the root, e.g. `/file.txt`. Is that desired, especially if the cleanup responsibility of the files is on the receiver? What if such a file (directory?) already exists on the filesystem, it looks like this would silently overwrite it? Would it make more sense to have a directory specific to GLFW file drops, e.g. `/.glfw_dropped_files/ + file.name.replace(/\//g,'_')` or something like that?